src/fs.zig

```diff
@@ -1488,61 +1440,15 @@ pub const PathName = struct {
             std.debug.assert(!strings.startsWith(_path, "/:/"));
             std.debug.assert(!strings.startsWith(_path, "\\:\\"));
         }
-
-        var path = _path;
-        var base = path;
-        var ext: []const u8 = undefined;
-        var dir = path;
-        var is_absolute = true;
-        const has_disk_designator = path.len > 2 and path[1] == ':' and switch (path[0]) {
-            'a'...'z', 'A'...'Z' => true,
-            else => false,
-        } and bun.path.isSepAny(path[2]);
-        if (has_disk_designator) {
-            path = path[2..];
-        }
-
-        var _i = bun.path.lastIndexOfSep(path);
-        while (_i) |i| {
-            // Stop if we found a non-trailing slash
-            if (i + 1 != path.len and path.len > i + 1) {
-                base = path[i + 1 ..];
-                dir = path[0..i];
-                is_absolute = false;
-                break;
-            }
-
-            // Ignore trailing slashes
-            path = path[0..i];
-
-            _i = bun.path.lastIndexOfSep(path);
-        }
-
-        // Strip off the extension
-        if (strings.lastIndexOfChar(base, '.')) |dot| {
-            ext = base[dot..];
-            base = base[0..dot];
-        } else {
-            ext = "";
-        }
-
-        if (is_absolute) {
-            dir = &([_]u8{});
-        }
-
-        if (base.len > 1 and bun.path.isSepAny(base[base.len - 1])) {
-            base = base[0 .. base.len - 1];
-        }
-
-        if (!is_absolute and has_disk_designator) {
-            dir = _path[0 .. dir.len + 2];
-        }
-
+        const isAbsolute = std.fs.path.isAbsolute(_path);
+        const dir = if (isAbsolute) &([_]u8{}) else std.fs.path.dirname(_path) orelse &([_]u8{});
+        const base = std.fs.path.basename(_path);
+        const filename = if (dir.len > 0) _path[dir.len + 1 ..] else _path;
         return PathName{
             .dir = dir,
             .base = base,
-            .ext = ext,
-            .filename = if (dir.len > 0) _path[dir.len + 1 ..] else _path,
+            .ext = std.fs.path.extension(base),
+            .filename = filename,
         };
```

Comment:
☝️ The deleted code above looked to be doing what the standard zig lib does. So I replaced it with the std lib to reduce gotchas. I kept the deviation in behavior though. I need a second look here though 🙏 .